clap-utils/src/keypair.rs

```diff
@@ -0,0 +1,78 @@
+use bip39::{Language, Mnemonic, Seed};
+use clap::values_t;
+use rpassword::prompt_password_stderr;
+use solana_sdk::signature::{
+    keypair_from_mnemonic_and_passphrase, keypair_from_seed, read_keypair_file, Keypair,
+    KeypairUtil,
+};
+use std::{
+    error,
+    io::{stdin, stdout, Write},
+};
+
+/// Reads user input from stdin to retrieve a mnemonic and passphrase for keypair derivation
+pub fn keypair_from_mnemonic(keypair_name: &str) -> Result<Keypair, Box<dyn error::Error>> {
+    let mnemonic_phrase =
+        prompt_password_stderr(&format!("[{}] mnemonic phrase: ", keypair_name)).unwrap();
+    match Mnemonic::from_phrase(mnemonic_phrase.trim(), Language::English) {
+        Ok(mnemonic) => {
+            let passphrase =
+                prompt_password_stderr(&format!("[{}] (optional) passphrase: ", keypair_name))
+                    .unwrap();
+            let seed = Seed::new(&mnemonic, &passphrase);
+            keypair_from_seed(seed.as_bytes())
+        }
+        Err(err) => {
+            print!(
+                "[{}] mnemonic phrase validation failed, continue anyways? (y/n): ",
```

Comment:
This message might need supplemented to suggest when it's OK to continue

Reply:
How does this look?

```
[identity-keypair] mnemonic phrase: 
Your mnemonic phrase failed to validate with the official BIP39 English word list.
If your phrase used a different word list, you can optionally continue: (y/n): y
[identity-keypair] (optional) passphrase: 
```

Reply:
> If your phrase used a different word list, you can optionally continue. (y/n): 

What would be a case where somebody would want to use a different word list? 

> [identity-keypair] (optional) passphrase: 

Should we wrap this behind an optional command-line argument requesting a passphrase?

Basically my concern about both these two "optional" settings is avoiding user confusion.  If I didn't use explicitly use a different word list or passphrase it seems like I could make a user error that would take me to a bad place.

Reply:
@mvines I just added an arg flag for skipping mnemonic validation to help avoid confusion on that part.

I feel that the prompt for an optional passphrase is pretty quick to skip though and could stay as is. It's a part of the BIP39 standard and in theory wouldn't be surprising to users who generated a mnemonic phrase already.

Speaking of which... https://github.com/solana-labs/solana/issues/7088

Reply:
Maybe we can improve the passphrase prompt with something like, "If this mnemonic has an associated passphrase, enter it now.  Otherwise, press ENTER to continue. : "?

Reply:
Also would calm my OCD with a minor refactor to only have one instance of the passphrase prompting in code 🙏

Reply:
@t-nelson yeah that's basically what the experience is now. At this prompt, you can press ENTER to continue with an empty string passphrase

```
[identity-keypair] (optional) passphrase:
```

Reply:
Sure.  I'm more concerned about the ambiguity of "optional". Hinting to the user that there MAY be a be an associated passphrase should clear this up in most cases.

Reply:
Ah got it! Sounds good 👍

Reply:
@t-nelson updated but kept the 2 separate instances of passphrase prompting due to different control flows for validated vs non-validated seed phrases. I believe it's cleaner as is than the following:

```rust
    let mnemonic = if !skip_validation {
        Some(Mnemonic::from_phrase(seed_phrase, Language::English)?)
    } else {
        None
    };
    
    let passphrase = prompt_password_stderr(&passphrase_prompt)?;
    if let Some(mnemonic) = mnemonic {
        let seed = Seed::new(&mnemonic, &passphrase);
        keypair_from_seed(seed.as_bytes())
    } else {
        keypair_from_seed_phrase_and_passphrase(&seed_phrase, &passphrase)
    }
```

Reply:
WFM.  You got the worst of it by moving the prompt string out to a single instance 😉

Reply:
haha had a feeling 😉